core-api/src/main/java/com/optimizely/ab/event/BatchEventProcessor.java

```diff
@@ -142,6 +147,12 @@ public void run() {
                         break;
                     }
 
+                    if (item == FLUSH_SIGNAL) {
+                        logger.debug("Received flush signal.");
+                        flush();
+                        break;
```

Comment:
Is it correct to break here? Then the event consumer will exit, and no more events will be consumed, right?

Reply:
Yikes. Yeah should be a `continue;` 🤦‍♂